lib/wibox/widget/base.lua

```diff
@@ -101,6 +101,15 @@ function base.make_widget(proxy)
         ret._fit_geometry_cache = setmetatable({}, { __mode = 'v' })
     end)
 
+    -- Add visible property and setter.
+    ret.visible = true
+    ret.set_visible = function(b)
+        if b ~= ret.visible then
+            ret.visible = b
+            ret:emit_signal("widget::updated")
+        end
+    end
+
```

Comment:
I'd make this a `proper method` instead of catching `ret` by an upvalue. So something like:
```
    function ret:set_visible(b)
        if b ~= self.visible then
            self.visible = b
            self:emit_signal("widget::updated")
        end
    end
```
Seems more consistent to me to other stuff "out there".